### PR TITLE
add test integration tests

### PR DESCRIPTION
**Pull request by paetling**
review @thieman I added the a handler which can be used in tests to get the output from client calls to log_to_client.  Using this I wrote integration tests testing the test cli command

### CODE REVIEW

File: tests/fixtures.py [hunk 1]
```diff
@@ -93,16 +93,26 @@ def specs_fixture_with_depends():
...
 
-def busybox_single_app_bundle_fixture(num_bundles=1, command=['sleep 999999999']):
+def busybox_single_app_bundle_fixture(num_bundles=1, command=['sleep 999999999'], include_tests=False):
     """Fixture for use in integration tests. The local repo at
     /tmp/fake-repo should be set up before using this fixture."""
+    app_dict = {'repo': '/tmp/fake-repo',
+                'mount': '/repo',
+                'image': 'busybox',
+                'commands': {'always': command}}
+    if include_tests:
```
Comment (thieman) [inline tests/fixtures.py]: Why does this need to be optional?

File: tests/integration/cli/test_test.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        self.assertEqual(self.handler.log_to_client_output.count('OK'), 1)
+        self.assertEqual(self.handler.log_to_client_output.count('Running commands to create new image:'), 1)
+
+    def test_basic_test_args(self):
+        try:
+            self.run_command('test --recreate busyboxa test1 test/location')
+        except:
+            pass
+        self.assertEqual(self.handler.log_to_client_output.count('TESTS FAILED'), 1)
+        self.assertEqual(self.handler.log_to_client_output.count('OSError: No such file'), 1)
```
Comment (thieman) [inline tests/integration/cli/test_test.py]: This is kinda jank, is there a more specific / less brittle way to test that your args did something?

File: tests/integration/cli/test_test.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+from ...testcases import DustyIntegrationTestCase
+from ...fixtures import busybox_single_app_bundle_fixture
+from dusty.log import client_logger, DustyClientTestingSocketHandler
+
+class TestTestCLI(DustyIntegrationTestCase):
+    def setUp(self):
+        super(TestTestCLI, self).setUp()
+        busybox_single_app_bundle_fixture(num_bundles=1, include_tests=True)
+        self.handler = DustyClientTestingSocketHandler()
+        client_logger.addHandler(self.handler)
```
Comment (thieman) [inline tests/integration/cli/test_test.py]: Should this be in the general setUp for DustyIntegrationTestCase?

File: tests/fixtures.py (hunk 1 above)
Comment (paetling) [inline tests/fixtures.py]: it probably does not, just did not want to step on other tests toes.

File: tests/integration/cli/test_test.py (hunk 2 above)
Comment (paetling) [inline tests/integration/cli/test_test.py]: yes.  my test2 could be something like `ls` and I could list a different location.

File: tests/integration/cli/test_test.py (hunk 3 above)
Comment (paetling) [inline tests/integration/cli/test_test.py]: interesting question. It does not technically need to be, it probably should be just to help people in the future.

File: tests/fixtures.py (hunk 1 above)
Comment (thieman) [inline tests/fixtures.py]: Would be simpler to not make it optional.

**Pull request merged**